letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+      // conditional transfer.
+      const secret = crypto.randomBytes(32)
+      const fulfillment = base64url(secret)
+      const condition = base64url(crypto.createHash('sha256').update(secret).digest())
+
+      //Get the letter that we are selling
+      const letter = ('ABCDEFGHIJKLMNOPQRSTUVWXYZ').split('')[(Math.floor(Math.random() * 26))]
+
+      console.log(`    - Generated letter (${letter}) for customer to collect at http://localhost:8000${req.url}${fulfillment}`)
+
+      //Store the fulfillment (indexed by condition) to use when we get paid
+      fulfillments[condition] = fulfillment
+
+      //Store the letter (indexed by the fulfillment) to use when the customer requests it
+      letters[fulfillment] = letter
+
+      console.log(`    - Waiting for payment...`)
+
+      res.end(`Please send an Interledger payment of ${normalizedCost} ${ledgerInfo.currencyCode} to ${account} using the condition ${condition}\n` +
+              `> node ./pay.js ${account} ${cost} ${condition}`)
+      
+    } else {
+
+      //Request for a letter with the fulfillment in the path
+
+      //Get fulfillment from the path
+      const fulfillment = requestUrl.path.substring(1)
+
+      //Lookup the letter we stored previously for this fulfillment
+      const letter = letters[fulfillment]
+  
+      if(!letter) {
+
+        //We have no record of a letter that was issued for this fulfillment
+        // Respond with a 404 HTTP Status Code (Not Found)
+        res.statusCode = 404;
+        
+        console.log(`    - No letter found for fulfillment: ${fulfillment}`)
+
+        res.end("Unrecognized fulfillment.")
+
+      } else {
+
+        // Provide the customer with their letter
+        res.end(`Your letter: ${letter}`)
+
+        console.log(` 5. Providing paid letter to customer for fulfillment ${fulfillment}`)
+
+      }
+
+    }
+  }).listen(8000, function(){
+    console.log(`    - Listening on http://localhost:8000`)
+    console.log(` 3. Visit http://localhost:8000 in your browser to buy a letter`)
+  })
+
+```
+
+This snippet defines two code paths for incoming HTTP requests (well three but the first is just a hack to ignore favicon requests).
+
+The first path handles requests to http://localhost:8000/ (i.e. no token in the URL) and defines the logic to generate a new letter, but to store this in memory until the customer pays for it.
+
+First we set the HTTP Response code to 402 (Payment Required). This is not necessary but it's a useful convention. We'll see later how this becomes more valuable.
+
+Then we generate the data required to request an Interledger payment from the customer.
+
+```js
+const secret = crypto.randomBytes(32) //Get a random 32-byte number as our secret
+const fulfillment = base64url(secret) //Encode it as a string so we can share it easily (put it in URLs etc.)
+const condition = base64url(crypto.createHash('sha256').update(secret).digest()) //Hash the secret to get the condition we'll use for the payment
```

Comment:
I updated the variable names in this code, since, it's now like this: https://github.com/interledger/tutorials/blob/master/shop.js#L52-L53